### DISCUSSION TITLE
I am trying to custom the yolo11x

### DISCUSSION THREAD

**Discussion by Dung-NV129** [ultralytics/ultralytics#17606] [Q&A]
This is my custom layers i want to change,  when I add these layer to yolo11.yaml
backbone:
  - [-1, 1, HGStem, [32, 64]] # 0-P2/4
  - [-1, 3, HGBlock, [64, 128, 3]] 
  - [-1, 1, DWConv, [128, 3, 2, 1, False]] # 2-P3/8
  - [-1, 3, HGBlock, [128, 256, 3]] 
  - [-1, 3, HGBlock, [128, 256, 3, False, True]] 
  - [-1, 1, DWConv, [256, 3, 2, 1, False]] # 5-P4/16
  - [-1, 3, HGBlock, [256, 512, 5, True, False]] 
  - [-1, 3, HGBlock, [256, 512, 5, True, True]] 
  - [-1, 1, DWConv, [512, 3, 2, 1, False]] # 8-P5/32
  - [-1, 2, HGBlock, [512, 1024, 5, True, False]] 
  - [-1, 2, HGBlock, [512, 1024, 5, True, True]] 

neck:
  - [-1, 1, SPPF, [1024, 5]] 
  - [-1, 1, Conv, [512, 1, 1, None, 1, 1, False]]
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [7, 1, Conv, [512, 1, 1, None, 1, 1, False]] 
  - [[-2, -1], 1, Concat, [1]]
  - [-1, 3, RepC3, [512]]
  - [-1, 1, Conv, [256, 1, 1, None, 1, 1, False]]
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [4, 1, Conv, [256, 1, 1, None, 1, 1, False]]
  - [[-2, -1], 1, Concat, [1]]
  - [-1, 3, RepC3, [256]]

head:
  - [-1, 1, Conv, [256, 3, 2]] 
  - [[-1, 9], 1, Concat, [1]] 
  - [-1, 3, RepC3, [512]] 
  - [-1, 1, Conv, [512, 3, 2]] 
  - [[-1, 6], 1, Concat, [1]] 
  - [-1, 3, RepC3, [1024]] 
  - [[6, 9, 12], 1, RTDETRDecoder, [nc]] 
but the summary just still summary the old model like below:
![image](https://github.com/user-attachments/assets/bd84d3e8-4d4a-482f-a3d1-a250512331b3)

What should i do to custom the yolo11x model?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Dung-NV129, thank you for reaching out and engaging with the Ultralytics community 🚀! It looks like you're diving into some exciting customizations of the YOLO11x model. For a deep-dive into modifying model layers, the [Docs](https://docs.ultralytics.com) are a great resource, particularly the sections on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, which can provide guidance on proper configurations and adjustments.

If this is a 🐛 Bug Report relating to your model summarization issue, please ensure you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This will help us understand the context and provide more accurate support.

If your query is a custom training ❓ question, please share additional details, such as your training dataset examples, any errors or logs you've encountered, and confirm you're adhering to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). This information will allow us to provide more targeted advice.

Join the Ultralytics community wherever you feel the most comfortable! For real-time discussions, head over to [Discord](https://ultralytics.com/discord) 🎧. If you prefer more extended discourse, check out [Discourse](https://community.ultralytics.com). Or for a laid-back community vibe, try our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you are using the latest version of the `ultralytics` package, complete with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). This can help verify that any bugs you are encountering are not fixed in the latest updates:

```bash
pip install -U ultralytics
```

## Environments

Here are some verified environments you might find useful for running YOLO with all dependencies pre-installed:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Refer to [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. Refer to [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Refer to [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, it means all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, verifying correct operations across macOS, Windows, and Ubuntu. 

Your patience is greatly appreciated as an Ultralytics engineer will jump in to assist you soon! 😊

**Comment by Dung-NV129** [comment]
I was change both of "yolo11.yaml" and "rtdetr-x.yaml" by the new structure with existing classes.
![image](https://github.com/user-attachments/assets/7539ea85-9855-4f5c-97d5-7ffc652b6d35)
But the summary was not change

**Reply by glenn-jocher** [reply]
To apply your custom changes to the YOLO11 model, ensure you've correctly modified the configuration file and saved it in the appropriate directory. After making these changes, restart your environment or reload the model to reflect the updates. If the issue persists, please verify that the modified configuration file is being referenced correctly in your code.

**Reply by Dung-NV129** [reply]
where is the location file to change the model? I was change backbone and head in yolo11.yaml and rtdetr-x.yaml but the summary is not changes.

**Reply by glenn-jocher** [reply]
@Dung-NV129 ensure that your modified YAML files are being correctly referenced in your training script. Double-check the file paths and confirm that the new configurations are being loaded instead of any default settings. If the issue persists, try restarting your environment to ensure no old configurations are cached.

**Reply by Dung-NV129** [reply]
When I try to improve the deep of yolo11 like below:
backbone:

  - [-1, 2, C3k2, [32, False, 0.25]]
  - [-1, 1, GhostConv, [32, 3, 2]] # 0-P0/2
  - [-1, 2, C3k2, [64, False, 0.25]]
  - [-1, 1, GhostConv, [64, 3, 2]] # 0-P1/2
  - [-1, 2, C3k2, [128, False, 0.25]]
  - [-1, 1, GhostConv, [128, 3, 2]] # 1-P2/4
  - [-1, 2, C3k2, [256, False, 0.25]]
  - [-1, 1, GhostConv, [256, 3, 2]] # 3-P3/8
  - [-1, 2, C3k2, [512, False, 0.25]]
  - [-1, 1, GhostConv, [512, 3, 2]] # 5-P4/16
  - [-1, 2, C3k2, [512, True]]
  - [-1, 1, GhostConv, [1024, 3, 2]] # 7-P5/32
  - [-1, 2, C3TR, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]] # 9
  - [-1, 2, C2fPSA, [1024]] # 10


head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 2, C3k2, [512, False]] # 13

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 2, C3k2, [256, False]] # 16 (P3/8-small)

  - [-1, 1, GhostConv, [256, 3, 2]]
  - [[-1, 13], 1, Concat, [1]] # cat head P4
  - [-1, 2, C3k2, [512, False]] # 19 (P4/16-medium)

  - [-1, 1, GhostConv, [512, 3, 2]]
  - [[-1, 10], 1, Concat, [1]] # cat head P5
  - [-1, 2, C3TR, [1024, True]] # 22 (P5/32-large)

  - [[16, 19, 22], 1, Detect, [nc]] # Detect(P3, P4, P5)
  And i got an error below:
![image](https://github.com/user-attachments/assets/ee20762f-da52-4809-9baf-90b92e22570b)
I know the error because the head have size 32 but the backbone want to get size 8, but i do not know how to changes the head. If you have the answer or documentation to do this project. Please help me, Thanks a lot!

**Reply by glenn-jocher** [reply]
It seems like there's a mismatch in the feature map sizes between your backbone and head layers. Ensure that the output feature map sizes of your backbone match the expected input sizes of your head layers by adjusting the strides and layer configurations. You might find it helpful to explore the YOLO11 architecture documentation [here](https://docs.ultralytics.com/models/yolo11) for more insights.
